skimage/morphology/_flood_fill.py

```diff
@@ -236,13 +234,20 @@ def flood(image, seed_point, *, selem=None, connectivity=None, tolerance=None):
     except TypeError:
         seed_point = (seed_point,)
 
+    seed_point_as_array = np.asarray_chkfinite(seed_point)
+    if not np.all(
+            (0 <= seed_point_as_array) & (seed_point_as_array < image.shape)):
+        raise IndexError("seed_point lies outside the image.")
```

Comment:
@FirefoxMetzger I'd prefer to give users the convenience of using negative indexing. Additionally, I don't think `chkfinite` is needed since the indices should be int anyway. I'd suggest:

```suggestion
    seed_point = tuple(np.asarray(seed_point) % working_image.shape)
```

... But that needs to go after the `working_image` definition, below.

Reply:
@jni I saw the suggestion in the issue. I think it's a bit strange to use `%`, because it allows you to specify numbers completely outside the image like `seed_point = 2.5 * np.asarray(image.shape, dtype=np.intp)`, but why not.

Reply:
Oh, also shouldn't it be `image.shape` instead of `working_image.shape` as we want to avoid the seed being located on the padding, which is what causes the segfault in the first place?

Reply:
> also shouldn't it be image.shape instead of working_image.shape as we want to avoid the seed being located on the padding, which is what causes the segfault in the first place?

... 🤦 of course! Well spotted. =)

> I think it's a bit strange to use %, because it allows you to specify numbers completely outside the image like `seed_point = 2.5 * np.asarray(image.shape, dtype=np.intp)`

Sure, and for those cases I wouldn't be opposed to a check that ensures that the seed is in `[-image.shape, image.shape)`, but I think that's a rather minor concern. I generally think undocumented behaviour is ok, as long as the failure mode is not catastrophic (such as a segfault! 😂).